### PR TITLE
Remember scroll position take 2

### PR DESCRIPTION
**Pull request by r-franke**
See the functionality described in #905 

Comments have been addressed and changes have been made accordingly.

Also includes the fix for #529.

`src/js/flows/submitSearch.js` had a noteworthy change:

https://github.com/r-franke/brim/blob/42a1153257596e112d784003e4a2e0dc412b8ecd/src/js/flows/submitSearch.js#L29-L35

Previously the `save` parameter of `submitSearch` was of type `bool` and defaulted to `true`. This was changed to an object with default: `{history: true, investigation: true}`. All the calls to `submitSearch` throughout the code have been updated. 

- If it called `submitSearch` with no parameters specified, that stayed the same, so it's still calling `submitSearch` and both attributes of `save` have been set to `true`. 
- If it called `submitSearch(false)` it has been changed to submitSearch({history: false, investigation: false}).

@jameskerr Commented in:
https://github.com/brimsec/brim/pull/905#discussion_r449241579

I think a lot of different calls need to be reviewed to fix this so I split this off into issue #928

### CODE REVIEW

**Comment by r-franke** [comment]
Flow seems to be tripping over the tests in `src/js/brim/entries.test.js`. The tests themselves run fine they all pass. 

All errors are of the form:

`Cannot call brim.entries with object literal bound to the first parameter because string [1] is incompatible with
SearchRecord [2] in array element of property entries.`

or

`Cannot call history.push with "a" bound to entry because string [1] is incompatible with SearchRecord [2].`

Should we rewrite the tests to satisfy flow?

**Comment by jameskerr** [comment]
As a side note, I was so confused about how to properly type the `entries` function so that it would work with more than one "types" of entires, like "string" and "SearchRecord", that I asked a question on stack overflow.

https://stackoverflow.com/questions/62940543/how-to-annotate-a-generic-closure-that-returns-an-object-with-methods-with-flow

File: src/js/brim/entries.js [hunk 1]
```diff
@@ -22,6 +23,10 @@ export default function({entries: initEntries, position}: Args) {
       }
       return this
     },
+    update(scrollPos: ScrollPosition) {
```
[the remark this answers is not in the document]
Comment (r-franke) [reply src/js/brim/entries.js:26]: Good point. Updated.

**Comment by r-franke** [comment]
Changed made. Ready for review.

**Comment by philrz** [comment]
Verified in Brim commit `93ffb60`.

I first circled back to Brim commit `5159520` that came right before this change, as I actually was not familiar with this limitation. Per the attached video, if I'd scrolled-to/highlighted a particular location in the events list, performed some other search, then hit the Back button, I'd be brought back to the "top" of the events list, _not_ the location I'd been scrolled to.

[Repro.zip](https://github.com/brimsec/brim/files/4954684/Repro.zip)

Now on `93ffb60`, when I do the same and then hit the Back button, I'm brought back to the same scroll position & highlighted event that I was at before I executed the additional search.

[Verify.zip](https://github.com/brimsec/brim/files/4954693/Verify.zip)

Thanks @r-franke!

**Pull request merged**